edb/pgsql/resolver/command.py

```diff
@@ -452,6 +469,192 @@ def _uncompile_insert_object_stmt(
     )
 
 
+@dataclasses.dataclass(kw_only=True, frozen=True, repr=False, eq=False)
+class UncompileOnConflict:
+    ql_unless_conflict: tuple[qlast.Expr | None, qlast.Expr | None]
+
+    update_name: Optional[str] = None
+    update_id: Optional[irast.PathId] = None
+    update_input: Optional[pgast.Query] = None
+    update_rel: Optional[pgast.Relation] = None
+
+
+def _uncompile_on_conflict(
+    stmt: pgast.InsertStmt,
+    sub: s_objtypes.ObjectType,
+    sub_table: context.Table,
+    ctx: Context,
+    stype_refs: dict[uuid.UUID, list[qlast.Set]]
+) -> Optional[UncompileOnConflict]:
```

Comment:
Add a big comment describing what the edgeql we are generating is and what we do with it?